### PR TITLE
Integration redirectURI core preset

### PR DESCRIPTION
**Pull request by TheIsrael1**
This PR allows the framework to configure core presets like redirectURI and make available for integrations,

This PR also accounts for  the possible necessity for the integration's authenticator REDIRECT_URI to be overidden

### CODE REVIEW

File: packages/core/src/framework.ts [hunk 1]
```diff
@@ -324,10 +328,14 @@ export class Framework {
 
   makeWebhookUrl = ({ event, name }: { name: string; event: string }) => {
     return encodeURI(
-      `${this?.config?.systemHostURL}${this?.config?.routeRegistrationPath}/webhook?name=${name}&event=${event}`
+      `${this?.config?.systemHostURL}${this.routes.webhook}?name=${name}&event=${event}`
     );
   };
 
+  makeRedirectURI = () => {
+    return encodeURI(`${this?.config?.systemHostURL}${this.routes.callback}`);
```
Comment (abettke) [inline packages/core/src/framework.ts]: I would use the URL constructor to build this to ensure it creates a valid URL.

File: packages/core/src/integration.ts [hunk 2]
```diff
@@ -71,6 +78,10 @@ export class Integration<T = unknown> {
     this.dataLayer = dataLayer;
   }
 
+  attachCorePresets({ corePresets }: { corePresets: CoreIntegrationPresets }) {
```
Comment (abettke) [inline packages/core/src/integration.ts]: Is it necessary to make this a method? It only does one assignment.

**Review by abettke — APPROVED** [review]

File: packages/core/src/integration.ts (hunk 2 above)
Comment (TheIsrael1) [reply packages/core/src/integration.ts]: yeah I belive it's not necessary, assigning it to the property directly now

File: packages/core/src/framework.ts (hunk 1 above)
Comment (TheIsrael1) [reply packages/core/src/framework.ts]: Yeah good Idea, update made

**Pull request merged**